### ISSUE TITLE
Send Cookstyle metadata using Telemetry

### ISSUE THREAD

**Issue by tyler-ball** [chef/cookstyle#427]
## Motivation

```
As a cookstyle developer,
I want to know the prevalence of cops that are triggering but lack autocorrection,
so that I can appropriately focus energy on building autocorrection functionality.

As a Chef Infra developer,
I want to know how often deprecations are still being detected by cookstyle in codebases,
so that I can determine the correct time to deprecate functionality instead of setting arbitrary dates.
```

## Specification

Once cookstyle has been updated with a formatter that logs metadata we need to send that to our telemetry collection service. This includes implementing the opt-in / opt-out telemetry behavior to cookstyle, as well as integrating that process into the Chef Workstation suite.

## Questions

1. How do users opt-in to telemetry for specific tools and/or the entire suite of tools? Perhaps for today we do not offer that level of granularity.

## Downstream Impact

Potentially Chef Workstation and the telemetry library if we want to offer granularity of opt-in/out.
